core/src/main/scala/monocle/std/Tree.scala

```diff
@@ -2,6 +2,8 @@ package monocle.std
 
 import monocle.function._
 import monocle.{Iso, Lens}
+import monocle.function.all._
+import monocle.std.stream._
```

Comment:
I think you're correct. `headOption` instance is need bring in scope.
but, I get erros....  (add `-Ywarn-unused-import` option)

```scala
[aoino@aoino-local Monocle]$ sbt clean compile
...
[error] /Users/aoino/git/Monocle/core/src/main/scala/monocle/std/Either3.scala:5: Unused import
[error] import scalaz.{\/, -\/, \/-}
[error]                ^
[error] /Users/aoino/git/Monocle/core/src/main/scala/monocle/std/Tree.scala:6: Unused import
[error] import monocle.std.stream._
[error]                           ^
[error] two errors found
[error] (core/compile:compileIncremental) Compilation failed
[error] Total time: 35 s, completed 2015/09/08 0:18:44
```
(sorry, Either3's unused import error is my bad..., I will fix it soon...)

I have tried some pattern.

- delete `import monocle.std.stream._` only
```scala
[aoino@aoino-local Monocle]$ sbt clean compile
...
[success] Total time: 50 s, completed 2015/09/08 0:20:42
```

- delete `with StreamOptics` in `StdInstances` trait only
```scala
[aoino@aoino-local Monocle]$ sbt clean compile
...
[success] Total time: 43 s, completed 2015/09/08 0:22:08
```

- delete `import monocle.std.stream._` and `with StreamOptics`.
```scala
[aoino@aoino-local Monocle]$ sbt clean compile
[error] /Users/aoino/git/Monocle/core/src/main/scala/monocle/std/Tree.scala:32: Could not find an instance of Cons[Stream[scalaz.Tree[A]],scalaz.Tree[A]], please check Monocle instance location policy to find out which import is necessary
[error]       case xs    => Tree.node(tree.rootLabel, headOption[Stream[Tree[A]], Tree[A]].modify(_set(newLeaf))(xs) )
[error]                                                         ^
[error] /Users/aoino/git/Monocle/core/src/main/scala/monocle/std/Tree.scala:47: Could not find an instance of Snoc[Stream[scalaz.Tree[A]],scalaz.Tree[A]], please check Monocle instance location policy to find out which import is necessary
[error]       case xs    => Tree.node(tree.rootLabel, lastOption[Stream[Tree[A]], Tree[A]].modify(_set(newLeaf))(xs) )
[error]                                                         ^
[error] two errors found
[error] (core/compile:compileIncremental) Compilation failed
[error] Total time: 33 s, completed 2015/09/08 0:23:52
```


maybe, `Cons[Stream[Tree[A]], Tree[A]]` implicit value is resolved in `StdInstances` trait....?